### ISSUE TITLE
Bug:  Focalboard keeps stopping to work!

### ISSUE THREAD

**Issue by arladmin** [mattermost/focalboard#4699]
The Mattermost Boards plugin was functional and then without any intervention, it stopped working.

This cycle has been going on repeat, for the past several months. Where i install the Boards plugin, they stop working, then i reset the plugin/uninstall them from the DB and reinstall them.

![image](https://user-images.githubusercontent.com/64059154/229976431-b95d0e7c-4a4a-4114-8b9b-f287172537f5.png)

**Error:**
```
cannot activate plugin: error initializing the DB: driver: mysql, message: failed when applying migration, command: apply_migration, originalError: Error 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '2e11450c0d1cbe01.focalboard_workspaces TO focalboard_teams' at line 1, query: 


			SET @stmt = (SELECT IF(
				(
				SELECT COUNT(table_name) FROM INFORMATION_SCHEMA.TABLES
				WHERE table_name = 'focalboard_teams'
				AND table_schema = '2e11450c0d1cbe01'
				) &gt; 0,
				'SELECT 1;',
				'RENAME TABLE 2e11450c0d1cbe01.focalboard_workspaces TO focalboard_teams;'
			));
			PREPARE renameTableIfNeeded FROM @stmt;
			EXECUTE renameTableIfNeeded;
			DEALLOCATE PREPARE renameTableIfNeeded;
		
			SET @stmt = (SELECT IF(
				(
				SELECT COUNT(column_name) FROM INFORMATION_SCHEMA.COLUMNS
				WHERE table_name = 'focalboard_blocks'
				AND table_schema = '2e11450c0d1cbe01'
				AND column_name = 'channel_id'
				) &gt; 0,
				'SELECT 1;',
				'ALTER TABLE 2e11450c0d1cbe01.focalboard_blocks CHANGE workspace_id channel_id varchar(36);'
			));
			PREPARE renameColumnIfNeeded FROM @stmt;
			EXECUTE renameColumnIfNeeded;
			DEALLOCATE PREPARE renameColumnIfNeeded;
		

			SET @stmt = (SELECT IF(
				(
				SELECT COUNT(column_name) FROM INFORMATION_SCHEMA.COLUMNS
				WHERE table_name = 'focalboard_blocks_history'
				AND table_schema = '2e11450c0d1cbe01'
				AND column_name = 'channel_id'
				) &gt; 0,
				'SELECT 1;',
				'ALTER TABLE 2e11450c0d1cbe01.focalboard_blocks_history CHANGE workspace_id channel_id varchar(36);'
			));
			PREPARE renameColumnIfNeeded FROM @stmt;
			EXECUTE renameColumnIfNeeded;
			DEALLOCATE PREPARE renameColumnIfNeeded;
		
			SET @stmt = (SELECT IF(
				(
				  SELECT COUNT(column_name) FROM INFORMATION_SCHEMA.COLUMNS
				  WHERE table_name = 'focalboard_blocks'
				  AND table_schema = '2e11450c0d1cbe01'
				  AND column_name = 'workspace_id'
				) &gt; 0,
				'ALTER TABLE 2e11450c0d1cbe01.focalboard_blocks DROP COLUMN workspace_id;',
				'SELECT 1;'
			));
			PREPARE dropColumnIfNeeded FROM @stmt;
			EXECUTE dropColumnIfNeeded;
			DEALLOCATE PREPARE dropColumnIfNeeded;
		

			SET @stmt = (SELECT IF(
				(
				  SELECT COUNT(column_name) FROM INFORMATION_SCHEMA.COLUMNS
				  WHERE table_name = 'focalboard_blocks_history'
				  AND table_schema = '2e11450c0d1cbe01'
				  AND column_name = 'workspace_id'
				) &gt; 0,
				'ALTER TABLE 2e11450c0d1cbe01.focalboard_blocks_history DROP COLUMN workspace_id;',
				'SELECT 1;'
			));
			PREPARE dropColumnIfNeeded FROM @stmt;
			EXECUTE dropColumnIfNeeded;
			DEALLOCATE PREPARE dropColumnIfNeeded;
		
			SET @stmt = (SELECT IF(
				(
				  SELECT COUNT(column_name) FROM INFORMATION_SCHEMA.COLUMNS
				  WHERE table_name = 'focalboard_blocks'
				  AND table_schema = '2e11450c0d1cbe01'
				  AND column_name = 'board_id'
				) &gt; 0,
				'SELECT 1;',
				'ALTER TABLE 2e11450c0d1cbe01.focalboard_blocks ADD COLUMN board_id varchar(36) ;'
			));
			PREPARE addColumnIfNeeded FROM @stmt;
			EXECUTE addColumnIfNeeded;
			DEALLOCATE PREPARE addColumnIfNeeded;
		

			SET @stmt = (SELECT IF(
				(
				  SELECT COUNT(column_name) FROM INFORMATION_SCHEMA.COLUMNS
				  WHERE table_name = 'focalboard_blocks_history'
				  AND table_schema = '2e11450c0d1cbe01'
				  AND column_name = 'board_id'
				) &gt; 0,
				'SELECT 1;',
				'ALTER TABLE 2e11450c0d1cbe01.focalboard_blocks_history ADD COLUMN board_id varchar(36) ;'
			));
			PREPARE addColumnIfNeeded FROM @stmt;
			EXECUTE addColumnIfNeeded;
			DEALLOCATE PREPARE addColumnIfNeeded;
		
UPDATE focalboard_blocks SET fields = JSON_SET(fields, '$.columnCalculations', JSON_OBJECT()) WHERE JSON_EXTRACT(fields, '$.columnCalculations') = JSON_ARRAY();

UPDATE focalboard_blocks b
  JOIN (
    SELECT id, JSON_EXTRACT(fields, '$.columnCalculations') as board_calculations from focalboard_blocks
    WHERE JSON_EXTRACT(fields, '$.columnCalculations') &lt;&gt; JSON_OBJECT()
  ) AS s on s.id = b.root_id
  SET fields = JSON_SET(fields, '$.columnCalculations', JSON_ARRAY(s.board_calculations))
  WHERE JSON_EXTRACT(b.fields, '$.viewType') = 'table'
  AND b.type = 'view';




CREATE TABLE IF NOT EXISTS focalboard_boards (
    id VARCHAR(36) NOT NULL PRIMARY KEY,

    
	
	insert_at DATETIME(6) NOT NULL DEFAULT NOW(6),

    team_id VARCHAR(36) NOT NULL,
    channel_id VARCHAR(36),
    created_by VARCHAR(36),
    modified_by VARCHAR(36),
    type VARCHAR(1) NOT NULL,
    title TEXT NOT NULL,
    description TEXT,
    icon VARCHAR(256),
    show_description BOOLEAN,
    is_template BOOLEAN,
    template_version INT DEFAULT 0,
    
    properties JSON,
    card_properties JSON,
    
    
    
    create_at BIGINT,
    update_at BIGINT,
    delete_at BIGINT
) DEFAULT CHARACTER SET utf8mb4;
			SET @stmt = (SELECT IF(
				(
				  SELECT COUNT(index_name) FROM INFORMATION_SCHEMA.STATISTICS
				  WHERE table_name = 'focalboard_boards'
				  AND table_schema = '2e11450c0d1cbe01'
				  AND index_name = 'idx_boards_team_id_is_template'
				) &gt; 0,
				'SELECT 1;',
				'CREATE INDEX idx_boards_team_id_is_template ON 2e11450c0d1cbe01.focalboard_boards (team_id, is_template);'
			));
			PREPARE createIndexIfNeeded FROM @stmt;
			EXECUTE createIndexIfNeeded;
			DEALLOCATE PREPARE createIndexIfNeeded;
		

			SET @stmt = (SELECT IF(
				(
				  SELECT COUNT(index_name) FROM INFORMATION_SCHEMA.STATISTICS
				  WHERE table_name = 'focalboard_boards'
				  AND table_schema = '2e11450c0d1cbe01'
				  AND index_name = 'idx_boards_channel_id'
				) &gt; 0,
				'SELECT 1;',
				'CREATE INDEX idx_boards_channel_id ON 2e11450c0d1cbe01.focalboard_boards (channel_id);'
			));
			PREPARE createIndexIfNeeded FROM @stmt;
			EXECUTE createIndexIfNeeded;
			DEALLOCATE PREPARE createIndexIfNeeded;
		

CREATE TABLE IF NOT EXISTS focalboard_boards_history (
    id VARCHAR(36) NOT NULL,

    
	
	insert_at DATETIME(6) NOT NULL DEFAULT NOW(6),

    team_id VARCHAR(36) NOT NULL,
    channel_id VARCHAR(36),
    created_by VARCHAR(36),
    modified_by VARCHAR(36),
    type VARCHAR(1) NOT NULL,
    title TEXT NOT NULL,
    description TEXT,
    icon VARCHAR(256),
    show_description BOOLEAN,
    is_template BOOLEAN,
    template_version INT DEFAULT 0,
    
    properties JSON,
    card_properties JSON,
    
    
    
    create_at BIGINT,
    update_at BIGINT,
    delete_at BIGINT,

    PRIMARY KEY (id, insert_at)
) DEFAULT CHARACTER SET utf8mb4;
  
  
  INSERT INTO focalboard_boards (
      SELECT B.id, B.insert_at, C.TeamId, B.channel_id, B.created_by, B.modified_by, C.Type,
                 COALESCE(B.title, ''),
                 COALESCE(JSON_UNQUOTE(JSON_EXTRACT(B.fields,'$.description')), ''),
                 JSON_UNQUOTE(JSON_EXTRACT(B.fields,'$.icon')),
                 COALESCE(JSON_EXTRACT(B.fields, '$.showDescription'), 'false') = 'true',
                 COALESCE(JSON_EXTRACT(B.fields, '$.isTemplate'), 'false') = 'true',
                 COALESCE(JSON_EXTRACT(B.fields, '$.templateVer'), 0),
                 '{}', JSON_EXTRACT(B.fields, '$.cardProperties'), B.create_at,
                 B.update_at, B.delete_at 
          FROM focalboard_blocks AS B
          INNER JOIN Channels AS C ON C.Id=B.channel_id
          WHERE B.type='board'
  );
  INSERT INTO focalboard_boards_history (
      SELECT B.id, B.insert_at, C.TeamId, B.channel_id, B.created_by, B.modified_by, C.Type,
                 COALESCE(B.title, ''),
                 COALESCE(JSON_UNQUOTE(JSON_EXTRACT(B.fields,'$.description')), ''),
                 JSON_UNQUOTE(JSON_EXTRACT(B.fields,'$.icon')),
                 COALESCE(JSON_EXTRACT(B.fields, '$.showDescription'), 'false') = 'true',
                 COALESCE(JSON_EXTRACT(B.fields, '$.isTemplate'), 'false') = 'true',
                 COALESCE(JSON_EXTRACT(B.fields, '$.templateVer'), 0),
                 '{}', JSON_EXTRACT(B.fields, '$.cardProperties'), B.create_at,
                 B.update_at, B.delete_at 
          FROM focalboard_blocks_history AS B
          INNER JOIN Channels AS C ON C.Id=B.channel_id
          WHERE B.type='board'
  );
  
UPDATE focalboard_blocks SET board_id=root_id WHERE board_id IS NULL OR board_id='';
UPDATE focalboard_blocks_history SET board_id=root_id WHERE board_id IS NULL OR board_id='';DELETE FROM focalboard_blocks WHERE type = 'board';
DELETE FROM focalboard_blocks_history WHERE type = 'board';
CREATE TABLE IF NOT EXISTS focalboard_board_members (
    board_id VARCHAR(36) NOT NULL,
    user_id VARCHAR(36) NOT NULL,
    roles VARCHAR(64),
    scheme_admin BOOLEAN,
    scheme_editor BOOLEAN,
    scheme_commenter BOOLEAN,
    scheme_viewer BOOLEAN,
    PRIMARY KEY (board_id, user_id)
) DEFAULT CHARACTER SET utf8mb4;
INSERT INTO focalboard_board_members (
    SELECT B.Id, CM.UserId, CM.Roles, TRUE, TRUE, FALSE, FALSE
    FROM focalboard_boards AS B
    INNER JOIN ChannelMembers as CM ON CM.ChannelId=B.channel_id
    WHERE CM.SchemeAdmin=True OR (CM.UserId=B.created_by)
);




			SET @stmt = (SELECT IF(
				(
				  SELECT COUNT(index_name) FROM INFORMATION_SCHEMA.STATISTICS
				  WHERE table_name = 'focalboard_board_members'
				  AND table_schema = '2e11450c0d1cbe01'
				  AND index_name = 'idx_board_members_user_id'
				) &gt; 0,
				'SELECT 1;',
				'CREATE INDEX idx_board_members_user_id ON 2e11450c0d1cbe01.focalboard_board_members (user_id);'
			));
			PREPARE createIndexIfNeeded FROM @stmt;
			EXECUTE createIndexIfNeeded;
			DEALLOCATE PREPARE createIndexIfNeeded;
		


```

## Edition and Platform

 - Edition: Personal Server / Mattermost Boards (plugin)
 - Version: [e.g. v0.15.0]  Mattermost 7.9.1 / Mattermost Boards v7.9.2
 - Browser and OS: [e.g. Chrome on Mac, Edge on Windows]

## Additional context (optional)

Add any other context about the problem here, and any notes about the severity:  **Sev 1** (Unable to interact with/use Boards at all)
* Sev 1: Affects critical functionality without a workaround
* Sev 2: Affects major functionality with a difficult or non-obvious workaround
* Sev 3: Affects minor, non-critical functionality

**Comment by arladmin** [comment]
Is there no support for this?

**Comment by arladmin** [comment]
@mgdelacroix Hi. Could you please provide some info regarding this?

**Comment by arladmin** [comment]
Is Mattermost actually planning to kill Focalboard (Mattermost Boards) in the near-term?

Maybe that would explain the level of instability of the project and unresponsiveness of the team!

**Comment by mgdelacroix** [comment]
Hi @arladmin. Apologies for the delay. This is [a known issue](https://github.com/mattermost/focalboard/issues/4631) when using dabase names that start with a digit, for which MySQL requires escaping, and it's fixed already for version `v7.9.3`.

Could you please try to run that version and let us know if there is any issue with the upgrade?

> Is Mattermost actually planning to kill Focalboard (Mattermost Boards) in the near-term?

Mattermost is [focusing on the in product version of Focalboard](https://github.com/mattermost/focalboard/discussions/4645), but it is not discontinuing the app.

**Comment by mgdelacroix** [comment]
Closing the issue as it should be solved with https://github.com/mattermost/focalboard/pull/4653, please feel free to reopen if necessary
